### PR TITLE
feat: team feature domain and github

### LINKED ISSUE #2427: [Feature][Team] update github user to account
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

[Feature][Team] update github user to account

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### LINKED ISSUE #2431: [Feature][Team] domain layer team feature
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Existing users is to be renamed to accounts and moved to domainlayer/crossdomain package, it contains registration records from different platforms
Add a new table users (id varchar,name,email) in domainlayer/crossdomain, it contains unique identities, each identity may have multiple accounts
Add a new table user_accounts in domainlayer/crossdomain, it contains the relationship between users and accounts, note that user_accounts is a One-To-Many relationship, the PK should be account_id
Add a new table teams in domainlayer/crossdomain.
Add a new table team_users in domainlayer/crossdomain, it contains the Many2Many relationship between users and teams.

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by abeizn**
# Summary

update github and gitee users to accounts.

### Does this close any open issues?
closed #2427
closed #2431 

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

**Review by mindlesscloud — APPROVED** [review]
LGTM

File: plugins/github/models/migrationscripts/init_schema_20220615.go [hunk 1]
```diff
@@ -68,7 +69,7 @@ func (u *initSchemas) Up(ctx context.Context, db *gorm.DB) error {
 		&archived.GithubIssueComment{},
 		&archived.GithubIssueEvent{},
 		&archived.GithubIssueLabel{},
-		&archived.GithubUser{},
+		&archived.GithubAccount{},
```
Comment (likyh) [inline plugins/github/models/migrationscripts/init_schema_20220615.go:72]: No. it should not change existing migration scripts.

Comment (likyh) [reply plugins/github/models/migrationscripts/init_schema_20220615.go:72]: Or change the version for it at least. And it should drop old table `GithubUser` here.

Comment (likyh) [reply plugins/github/models/migrationscripts/init_schema_20220615.go:72]: image

**Pull request merged**